File: pydss/general/CNDataObjImpl.py
from dss.general.CNDataObj import (CNDataObj,)
from dss.general.impl.CableDataObjImpl import (CableDataObjImpl,)
from dss.general.CNData import (CNData,)


class CNDataObjImpl(CableDataObjImpl, CNDataObj):
    kStrand = None
    diaStrand = None
    gmrStrand = None
    rStrand = None

    def __init__(self, parClass, CNDataName):
        super(CNDataObjImpl, self)(parClass, CNDataName)
        self.setName(CNDataName.toLowerCase())
        self.objType = parClass.getDSSClassType()
        self.kStrand = 2
        self.diaStrand = -1.0
        self.gmrStrand = -1.0
        self.rStrand = -1.0
        self.initPropertyValues(0)

    def dumpProperties(self, f, complete):
        super(CNDataObjImpl, self).dumpProperties(f, complete)
        _0 = True
        i = 0
        while True:
            if _0 is True:
                _0 = False
            else:
                i += 1
            if not (i < self.getParentClass().getNumProperties()):
                break
            f.print_('~ ' + self.getParentClass().getPropertyName()[i] + '=')
            _1 = i
            _2 = False
            while True:
                if _1 == 0:
                    _2 = True
                    print String.format.format('%d', self.kStrand)
                    break
                if (_2 is True) or (_1 == 1):
                    _2 = True
                    print String.format.format('%.6g', self.diaStrand)
                    break
                if (_2 is True) or (_1 == 2):
                    _2 = True
                    print String.format.format('%.6g', self.gmrStrand)
                    break
                if (_2 is True) or (_1 == 3):
                    _2 = True
                    print String.format.format('%.6g', self.rStrand)
                    break
                break

    def initPropertyValues(self, arrayOffset):
        self.setPropertyValue(0, '2')
        self.setPropertyValue(1, '-1')
        self.setPropertyValue(2, '-1')
        self.setPropertyValue(3, '-1')
        super(CNDataObjImpl, self).initPropertyValues(arrayOffset + CNData.NumPropsThisClass)

    def getkStrand(self):
        return self.kStrand

    def getDiaStrand(self):
        return self.diaStrand

    def getGmrStrand(self):
        return self.gmrStrand

    def getRStrand(self):
        # FIXME: Private members in OpenDSS
        return self.rStrand

    def setkStrand(self, kstrand):
        self.kStrand = kstrand

    def setDiaStrand(self, diastrand):
        self.diaStrand = diastrand

    def setGmrStrand(self, gmrstrand):
        self.gmrStrand = gmrstrand

    def setRStrand(self, rstrand):
        self.rStrand = rstrand

File: core/jdss/src/main/java/com/epri/dss/general/impl/CNDataObjImpl.java
package com.epri.dss.general.impl;

import java.io.PrintStream;

import com.epri.dss.common.DSSClass;
import com.epri.dss.general.CNData;
import com.epri.dss.general.CNDataObj;

public class CNDataObjImpl extends CableDataObjImpl implements CNDataObj {

	private int kStrand;
	private double diaStrand;
	private double gmrStrand;
	private double rStrand;

	public CNDataObjImpl(DSSClass parClass, String CNDataName) {
		super(parClass, CNDataName);
		setName(CNDataName.toLowerCase());
		objType = parClass.getDSSClassType();

		kStrand   = 2;
		diaStrand = -1.0;
		gmrStrand = -1.0;
		rStrand   = -1.0;

		initPropertyValues(0);
	}

	@Override
	public void dumpProperties(PrintStream f, boolean complete) {
		super.dumpProperties(f, complete);

		for (int i = 0; i < getParentClass().getNumProperties(); i++) {
			f.print("~ " + getParentClass().getPropertyName()[i] + "=");
			switch (i) {
			case 0:
				f.println(String.format("%d", kStrand));
				break;
			case 1:
				f.println(String.format("%.6g", diaStrand));
				break;
			case 2:
				f.println(String.format("%.6g", gmrStrand));
				break;
			case 3:
				f.println(String.format("%.6g", rStrand));
				break;
			}
		}
	}

	@Override
	public void initPropertyValues(int arrayOffset) {
		setPropertyValue(0, "2");
		setPropertyValue(1, "-1");
		setPropertyValue(2, "-1");
		setPropertyValue(3, "-1");
		super.initPropertyValues(arrayOffset + CNData.NumPropsThisClass);
	}

	public int getkStrand() {
		return kStrand;
	}

	public double getDiaStrand() {
		return diaStrand;
	}

	public double getGmrStrand() {
		return gmrStrand;
	}

	public double getRStrand() {
		return rStrand;
	}

	// FIXME: Private members in OpenDSS

	public void setkStrand(int kstrand) {
		this.kStrand = kstrand;
	}

	public void setDiaStrand(double diastrand) {
		diaStrand = diastrand;
	}

	public void setGmrStrand(double gmrstrand) {
		gmrStrand = gmrstrand;
	}

	public void setRStrand(double rstrand) {
		rStrand = rstrand;
	}

}
